xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -1333,6 +1333,12 @@ void CXBMCApp::onReceive(CJNIIntent intent)
       if (winSystem && dynamic_cast<CWinSystemAndroid*>(winSystem))
         dynamic_cast<CWinSystemAndroid*>(winSystem)->SetHdmiState(hdmiPlugged);
     }
+    if (hdmiPlugged)
+    {
+      CLog::Log(LOGDEBUG, "{}: Reset audio engine", __FUNCTION__);
+      CServiceBroker::GetActiveAE()->Suspend();
+      CServiceBroker::GetActiveAE()->Resume();
```

Comment:
CServiceBroker::GetActiveAE()->DeviceChange(); should also work and semantically do the right thing. Suspend / Resume works by chance as Resume causes an internal reenumeration.

Bonus-Points:
As we basically only want to rescan when reinitialize AE when really something changed, you could compare the information you get with the HDMI_PLUG callback to see if something new pops up and only then issue the CServiceBroker::GetActiveAE()->DeviceChange(); call.

Reply:
That being said: Ignore the Bonus Points for now ;-). We can see if it's needed when reports on the forum popup. Only change the Suspend / Resume please to DeviceChange(); thanks!